subsys/net/ip/ipv6_nbr.c

```diff
@@ -664,8 +664,10 @@ enum net_verdict net_ipv6_prepare_for_send(struct net_pkt *pkt)
 	 */
 	if (net_pkt_ipv6_fragment_id(pkt) == 0) {
 		size_t pkt_len = net_pkt_get_len(pkt);
+		u16_t mtu;
 
-		if (pkt_len > NET_IPV6_MTU) {
+		mtu = MAX(NET_IPV6_MTU, net_if_get_mtu(net_pkt_iface(pkt)));
+		if (mtu < pkt_len) {
```

Comment:
I really think `pkt_len > mtu` is clearer here.

Reply:
Please use temp variable for storing iface mtu so that the function would not be called twice.

Reply:
well, mtu can be set to iface's mtu by default. No need of an extra variable

Reply:
MAX is defined as `#define MAX(a, b) (((a) > (b)) ? (a) : (b))`. If you place function call as a parameter, then in worst case, the function will be called twice. In this case it is not a big issue as the value is const, but it is a useless call to make here.